benchexec/tools/sv_benchmarks_util.py

```diff
@@ -40,3 +44,196 @@ def get_data_model_from_task(task, param_dict):
+    Non-witness files consist of all files which do not match the witness file name.
+    The witness file is identified by the option "witness" in the task options.
+
+    @param task: An instance of a task
+    @return: Unique non-witness file
+    """
+    other_files = get_non_witness_input_files(task)
+    if len(other_files) > 1:
+        raise UnsupportedFeatureException(
+            "Tool does not support tasks with more than one input file"
+        )
+    return other_files[0]
+
+
+def get_witness_options(options, task, witness_options):
+    """
+    This function returns the additional options to handle witnesses.
+    It checks if the witness is passed as an option or through the task definition.
+    If the witness is passed through both, it raises an exception.
+    If no witness is given in the task, it returns an empty list.
+    Therefore, this function can be used with any task regardless if it
+    has a witness option or not.
+
+    @param options: List of existing options
+    @param task: An instance of a task
+    @param witness_options: List of options which need to be set to handle witnesses
+        e.g. if the options are ["-w"], then the witness should be passed as "-w witness_file"
+        to the tool
+
+    @return: List of additional options to handle witnesses. They are constructed
+        based on witness_options. For example if witness_options is ["-w"], then
+        the return value will be ["-w", "witness_file"]
+    """
+
+    additional_options = []
+    if isinstance(task.options, dict) and "witness" in task.options.keys():
+        if not any(witness_option in options for witness_option in witness_options):
+            for witness_option in witness_options:
+                additional_options += [witness_option, get_witness(task)]
+        else:
+            raise benchexec.tools.template.UnsupportedFeatureException(
+                "You are passing a witness as both an option and through the task definition. "
+                "Please remove one of them."
+            )
+    return additional_options
+
+
+class TaskFilesConsidered(Enum):
+    """
+    Enum to represent the different types of input
+    files that can be considered for a task.
+    """
+
+    INPUT_FILES_OR_IDENTIFIER = 1
+    INPUT_FILES = 2
+    SINGLE_INPUT_FILE = 3
+
+
+def handle_witness_of_task(
+    task, options, witness_options, task_files_considered: TaskFilesConsidered
+):
+    """
+    This function returns the input files and witness options for a task.
+    The input files are based on the task_files_considered parameter.
+    The witness options are based on the witness_options parameter.
+
+    In validation task definition files, there is a key "witness" inside the
+    options of a task. To handle this type of tasks correctly in the tool info module
+    you should do the following two steps, if your tool does not already support
+    them natively:
```

Comment:
What do you mean with "support natively"? A witness validator in the context of SV-COMP never even gets the task-definition files.

Reply:
My goal was to indicate that tool developers could interpret the options in the task internally instead of changing their command line. I made this more precise in 15eba32e7cd6a91604fad75e91f449b14c865765

Reply:
Sorry, but "if your tool does not already support them by parsing the options correctly" still reads if there was a way for the tool to parse the options (in this context these are the options defined in the task definition) directly. The subject of the sentence would need to be `your tool-info module`. And it is better to use `task options`, because tool-info modules also get the options from the benchmark definition. Or just remove everything after the comma, I do not think it adds much value.

Reply:
Makes sense, the comment was deleted in 944d21e0f00de377ed952a1e1a8ab4805457d365

Reply:
Right, overlooked.